File: client/clojure/src/mednet_edc_api/specs/query_comment.clj
(ns mednet-edc-api.specs.query-comment
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def query-comment-data
  {
   (ds/opt :sequence) int?
   (ds/opt :annotationStatus) string?
   (ds/opt :user) string?
   (ds/opt :comment) string?
   (ds/opt :closed) boolean?
   (ds/opt :date) string?
   })

(def query-comment-spec
  (ds/spec
    {:name ::query-comment
     :spec query-comment-data}))

File: src/App/DTO/QueryComment.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

class QueryComment
{
    /**
     * Sequence number of the comment/action in the query history
     * @DTA\Data(field="sequence", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"int"})
     */
    public ?int $sequence = null;

    /**
     * Status of the query after this comment (e.g., Open, Closed)
     * @DTA\Data(field="annotationStatus", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $annotation_status = null;

    /**
     * Username of the user who made the comment or action
     * @DTA\Data(field="user", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $user = null;

    /**
     * Text of the comment
     * @DTA\Data(field="comment", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $comment = null;

    /**
     * Whether the query was marked closed at this step
     * @DTA\Data(field="closed", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"bool"})
     */
    public ?bool $closed = null;

    /**
     * Date of the comment or action
     * @DTA\Data(field="date", nullable=true)
     * @DTA\Validator(name="Scalar", options={"type":"string"})
     */
    public ?string $date = null;

}
